emscripten-simulated-jerry-api/jerry.js

```diff
@@ -148,6 +151,81 @@
       this._jerryToHostValueMap = {};
       this._nextObjectRef = 1;
       this._jerryInternalPropsWeakMap = new WeakMap();
+
+      if (!Module.Promise) {
+        console.warn('Please set Module.Promise to a promise implementation to use for JerryScript')
+      } else {
+        // Patch up the Promise.prototype then/catch implementations.
+        // The patched versions should create a function that when called, pushes the original callback arg to the jobQueue
```

Comment:
Maybe add a `var isPromisePatched` in the outer scope (same scope level as `var global = ...` up at the top of the file). I think if the client would call `jerry_init()` again, we are now patching it again.